src/index.ts

```diff
@@ -40,6 +52,40 @@ const createGenerator: CodegenGeneratorConstructor = (config, context) => {
 			return !!value.mediaType.mimeType.match('\\bjson$')
 		})
 
+		hbs.registerHelper('hasOperationSupportingMultipart', function(value: CodegenOperation | CodegenOperation[]): boolean {
+			if (isCodegenOperation(value)) {
+				return containsMultipartOperation([value])
+			} else {
+				return containsMultipartOperation(value)
+			}
+		})
+
+		hbs.registerHelper('toFileUploadProperties', function(properties: CodegenProperties): FileUploadProperty[] {
+			const results: FileUploadProperty[] = []
+			for (const prop in properties) {
+				const property = properties[prop]
+				if (property.schema.purpose !== CodegenSchemaPurpose.METADATA) {
+					continue
+				}
+
+				if (isCodegenArraySchema(property.schema) && property.schema.component) {
+					results.push({
+						name: prop,
+						minCount: property.schema.minItems,
+						maxCount: property.schema.maxItems,
+					})
+				} else {
```

Comment:
And here that `property.schema.schemaType === CodegenSchemaType.FILE`

Reply:
Similar idea with this one except we won't need to deal with the `component`.